tutorials/tts/FastPitch_MultiSpeaker_Pretraining.ipynb

```diff
@@ -0,0 +1,867 @@
+    "for m in tqdm(valid_datas): get_pitch(m)\n",
+    "\n",
+    "train_pitchs = np.concatenate(train_pitchs)\n",
+    "pitch_mean = float(np.mean(train_pitchs))\n",
+    "pitch_std = float(np.std(train_pitchs))\n",
+    "\n",
+    "with open(os.path.join(manidir, 'pitch_stats.json'), 'w') as f:\n",
+    "    json.dump({'pitch':[pitch_mean, pitch_std]}, f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a41ee532",
+   "metadata": {},
+   "source": [
+    "## c. Training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7f954beb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p {confdir} && cd {confdir} && wget https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/examples/tts/conf/fastpitch_align_44100_adapter.yaml"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e82e6860",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p {codedir} && cd {codedir} && wget https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/examples/tts/fastpitch.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12307264",
+   "metadata": {},
+   "source": [
+    "### Important notes\n",
+    "* [Data] **speaker_id** in **sup_data_types**: each data has an unique speaker index (start from 0) in the input.\n",
+    "* [Data] **reference_audio** in **sup_data_types**: each data has a reference audio (from the same speaker) in the input.\n",
+    "* [Speaker] **model.speaker_encoder.lookup_module**: model creates lookup table to get speaker embedding from speaker id.\n",
+    "* [Speaker] **model.speaker_encoder.lookup_module.n_speakers**: model gets the speaker size. \n",
+    "* [Speaker] **model.speaker_encoder.gst_module**: model creates global style token to extract speaker information from reference audio.\n",
+    "* [Condition] **condition_types=\"['add', 'layernorm']\"**: insert conditions with `add` operation to inputs and `layernorm` operation to layernorms."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6ffdce37",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Normally 200 epochs\n",
+    "!(python {codedir}/fastpitch.py \\\n",
+    "--config-path={os.path.abspath(confdir)} \\\n",
+    "--config-name=fastpitch_align_44100_adapter.yaml \\\n",
+    "+init_from_pretrained_model=\"tts_en_fastpitch\" \\\n",
+    "sample_rate={sample_rate} \\\n",
+    "train_dataset={train_manifest} \\\n",
+    "validation_datasets={valid_manifest} \\\n",
+    "sup_data_types=\"['align_prior_matrix', 'pitch', 'speaker_id','reference_audio']\" \\\n",
+    "sup_data_path={suppdir} \\\n",
+    "pitch_mean={pitch_mean} \\\n",
+    "pitch_std={pitch_std} \\\n",
+    "phoneme_dict_path={normdir}/cmudict-0.7b_nv22.10 \\\n",
+    "heteronyms_path={normdir}/heteronyms-052722 \\\n",
+    "model.speaker_encoder.lookup_module._target_=\"nemo.collections.tts.modules.submodules.SpeakerLookupTable\" \\\n",
```

Comment:
`phoneme_dict_path` and `heteronyms_path ` are removed

Reply:
`phoneme_dict_path `and `heteronyms_path ` are needed because they are dependent on where the user stores the file.

I meant the model.speaker_encoder.*, model.input_fft.*, etc are providing the same values as in the .yaml config, and so do nothing.

Reply:
`model.speaker_encoder` and `model.input_fft.condition_types` are new arguments. I am wondering if I don't set here, do user know how these parameters work?

Reply:
Well, the users will not understand what those parameters do anyways unless the tutorial explains it.  If we want to explain what they do and how a user might adjust them based on their dataset, then we can explain it in bullet points in a text section similar to https://github.com/NVIDIA/NeMo/blob/main/tutorials/tts/FastPitch_MixerTTS_Training.ipynb?short_path=558c0d9#L520-L525

Alternatively, you could explain it by pulling the values into python variables ahead of time and explaining in comments.

```
# Change this depending on your GPU memory
train_batch_size = 8
# Set this to["add"] if you want to disable layer norm adaptation.
adapter_norm_type = ["add", "layernorm"]

...
model.train_ds.dataloader_params.batch_size={train_batch_size}
model.input_fft.condition_types={adapter_norm_type}
model.output_fft.condition_types={adapter_norm_type}
...
```

Reply:
I explain the parameters in a separate markdown block.